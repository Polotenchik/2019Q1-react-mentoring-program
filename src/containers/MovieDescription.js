import React from 'react';
import { getMoives } from '../data';
import { Title } from '../common/Title';
import { Rating } from '../common/Rating';
import { Label } from '../common/Label';

const movie = getMoives().data[0];

export const MovieDescription = () => {
    return (
        <div className="film">
            <div className="film-poster">
                <img src={movie['poster_path']} />
            </div>
            <div className='film-info'>
                <div className='film-header' >
                    <div>
                        <Title content={ movie['title']} />
                    </div>
                    <div>
                        <Rating value ={ +movie['vote_average'] }/>
                    </div>
                </div>
                <p>{ movie['tagline'] }</p>
                <div className='year-min'>
                    <Label content={ movie['release_date'] ? movie['release_date'].slice(0,4): movie['release_date'] } />
                    <Label content={ movie['overview'] ? movie['overview'] + ' min' : movie['overview'] } />
                </div>
                <p>{ movie['overview']}</p>
            </div>
        </div>
    );
};