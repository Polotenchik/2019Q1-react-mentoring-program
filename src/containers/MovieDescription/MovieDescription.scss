.movie-description {
    display: flex;
    margin: $elementPadding 0;
    justify-content: space-between;
    align-items: center;
    background: rgba(0,0,0,0.5);
}

.movie-description-poster img {
    width: $filmPosterWidth;
}

.movie-description-info {
    width: $filmInfoWidth;
    padding: 0 $elementPadding 0 0;  
}

.movie-description-info-header {
    display: flex;
    align-items: end;
    cursor: default;
}

.caption {
    font-size: $textSize;
    line-height: 25px;
    padding: $elementPadding 0;
    text-align: justify;
    color: $color_white;
    font-style: italic;
}

.year-min {
    display: flex;
}
  
.year-min > div {
    margin: 0 $elementPadding*2 $elementPadding $elementPadding*2
}
  
.year-min > div:first-child {
    margin: 0 0 $elementPadding 0
}
  
.year-min > div:last-child {
    margin: 0 0 $elementPadding $elementPadding*2
}