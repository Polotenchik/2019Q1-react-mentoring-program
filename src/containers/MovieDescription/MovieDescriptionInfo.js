import React from 'react';
import PropTypes from 'prop-types';
import { Title, Rating, Label } from '../../common';

export const MovieDescriptionInfo = ({ info }) => {
    return (
        <div className='movie-description-info'>
            <div className='movie-description-info-header' >
                <Title content={ info['title']} ttlRed ttlBig />
                <Rating value ={ +info['vote_average'] }/>
            </div> 
            <p className='caption'>{ info['tagline'] }</p>
            <div className='year-min'>
                <Label labelWhite labelBold
                    content={ info['release_date'] 
                    ? info['release_date'].slice(0,4) 
                    : info['release_date'] }
                />
                <Label labelWhite labelBold
                    content={ info['runtime'] 
                    ? info['runtime'] + ' min' 
                    : info['runtime'] } 
                />
            </div>
            <p className='caption'>{ info['overview'] }</p>
        </div>
    );
};
        