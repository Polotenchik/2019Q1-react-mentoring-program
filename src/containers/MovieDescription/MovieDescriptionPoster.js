import React from 'react';

export const MovieDescriptionPoster = ({ posterPath }) => {

    return (
        <div className="film-poster">
            <img src={posterPath} />
        </div>  
    );
}
    
