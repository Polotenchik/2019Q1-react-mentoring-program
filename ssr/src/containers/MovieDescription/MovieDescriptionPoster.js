import React from 'react';

export const MovieDescriptionPoster = ({ posterPath }) => (
        <div className="film-poster">
            <img src={ posterPath } />
        </div>
);
