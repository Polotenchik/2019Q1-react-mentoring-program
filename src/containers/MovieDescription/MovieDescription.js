import React from 'react';
import PropTypes from 'prop-types';
import { MovieDescriptionInfo } from './MovieDescriptionInfo';
import { MovieDescriptionPoster } from './MovieDescriptionPoster'

export const MovieDescription = ({ active, info }) => {
    if (active && info) {
       return ( 
            <div className="movie-description">
                <MovieDescriptionPoster posterPath={ info['poster_path'] } />
                <MovieDescriptionInfo info={ info } />
            </div>
        ); 
    } else {
        return null;
    }
};

MovieDescription.defaulProps = {
    active: false,
    info: { }
};