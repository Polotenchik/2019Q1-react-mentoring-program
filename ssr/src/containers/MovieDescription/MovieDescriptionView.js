import React from 'react';
import PropTypes from 'prop-types';
import { MovieDescriptionInfo } from './MovieDescriptionInfo';
import { MovieDescriptionPoster } from './MovieDescriptionPoster';
import { Whooops404 } from '../../components';


export class MovieDescriptionView extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    if (this.props.movieMode && this.props.currentMovie) {
      const obj = Object.keys(this.props.currentMovie);
      if (obj.length === 0) {
        return (
                    <div className={'no-found'}>
                        Resource not found
                    </div>
        );
      }

      return (
                <div className="movie-description">
                    <MovieDescriptionPoster posterPath={ this.props.currentMovie.poster_path } />
                    <MovieDescriptionInfo info={ this.props.currentMovie } />
                </div>
      );
    }
    return null;
  }
}


MovieDescriptionView.defaulProps = {
  movieMode: false,
  info: { },
};
