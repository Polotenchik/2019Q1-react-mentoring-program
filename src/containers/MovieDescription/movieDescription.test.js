import React from 'react';
import { MovieDescription } from './MovieDescription';
import { shallow } from 'enzyme';

const mockMovieId =  337167;

describe('<MovieDescription />', () => {
  describe('Rendering', () => {
    it('should render MovieDescription', () => {
      const wrapper = shallow(<MovieDescription movie={ mockMovieId } />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});