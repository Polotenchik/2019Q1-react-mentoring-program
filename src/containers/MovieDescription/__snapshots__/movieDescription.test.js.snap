// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieDescription /> Rendering should render MovieDescription 1`] = `""`;
